sql/core/src/test/scala/org/apache/spark/sql/execution/python/PythonStreamingDataSourceSuite.scala

```diff
@@ -230,4 +273,220 @@ class PythonStreamingDataSourceSuite extends PythonDataSourceSuiteBase {
       stream => stream.commit(offset)
     }
   }
+
+  Seq("append", "complete").foreach { mode =>
+    test(s"data source stream write - $mode mode") {
+      assume(shouldTestPandasUDFs)
+      val dataSource =
+        createUserDefinedPythonDataSource(dataSourceName, simpleDataStreamWriterScript)
+      spark.dataSource.registerPython(dataSourceName, dataSource)
+      val inputData = MemoryStream[Int]
+      withTempDir { dir =>
+        val path = dir.getAbsolutePath
+        val checkpointDir = new File(path, "checkpoint")
+        checkpointDir.mkdir()
+        val outputDir = new File(path, "output")
+        outputDir.mkdir()
+        val streamDF = if (mode == "append") {
+          inputData.toDF()
+        } else {
+          // Complete mode only supports stateful aggregation
```

Comment:
Forgive my limited knowledge of streaming, but what does 'complete' mode mean here? Is it the same as setting 'overwrite=true'?

Reply:
My understanding is that Complete mode is equivalent to 'overwrite=true'. Correct me if I am wrong. @HeartSaVioR

Reply:
complete mode requires sink to support truncate.